livekit-android-sdk/src/main/java/io/livekit/android/room/participant/LocalParticipant.kt

```diff
@@ -763,6 +776,369 @@ internal constructor(
+     * and they will be received on the caller's side with the message intact.
+     * Other errors thrown in your handler will not be transmitted as-is, and will instead arrive to the caller as `1500` ("Application Error").
+     *
+     * @param method The name of the indicated RPC method
+     * @param handler Will be invoked when an RPC request for this method is received
+     */
+    @Suppress("RedundantSuspendModifier")
+    suspend fun registerRpcMethod(
+        method: String,
+        handler: RpcHandler,
+    ) {
+        this.rpcHandlers[method] = handler
+    }
+
+    /**
+     * Unregisters a previously registered RPC method.
+     *
+     * @param method The name of the RPC method to unregister
+     */
+    fun unregisterRpcMethod(
+        method: String,
+    ) {
+        this.rpcHandlers.remove(method)
+    }
+
+    internal fun handleDataPacket(packet: DataPacket) {
+        when {
+            packet.hasRpcRequest() -> {
+                val rpcRequest = packet.rpcRequest
+                scope.launch {
+                    handleIncomingRpcRequest(
+                        callerIdentity = Identity(packet.participantIdentity),
+                        requestId = rpcRequest.id,
+                        method = rpcRequest.method,
+                        payload = rpcRequest.payload,
+                        responseTimeoutMs = rpcRequest.responseTimeoutMs,
+                        version = rpcRequest.version,
+                    )
+                }
+            }
+
+            packet.hasRpcResponse() -> {
+                val rpcResponse = packet.rpcResponse
+                var payload: String? = null
+                var error: RpcError? = null
+
+                if (rpcResponse.hasPayload()) {
+                    payload = rpcResponse.payload
+                } else if (rpcResponse.hasError()) {
+                    error = RpcError.fromProto(rpcResponse.error)
+                }
+                handleIncomingRpcResponse(
+                    requestId = rpcResponse.requestId,
+                    payload = payload,
+                    error = error,
+                )
+            }
+
+            packet.hasRpcAck() -> {
+                val rpcAck = packet.rpcAck
+                handleIncomingRpcAck(rpcAck.requestId)
+            }
+        }
+    }
+
+    /**
+     * Initiate an RPC call to a remote participant
+     * @param destinationIdentity The identity of the destination participant.
+     * @param method The method name to call.
+     * @param payload The payload to pass to the method.
+     * @param responseTimeoutMs Timeout for receiving a response after initial connection (milliseconds). Defaults to 10000.
+     * @return The response payload.
+     * @throws RpcError on failure. Details in [RpcError.message].
+     */
+    suspend fun performRpc(
+        destinationIdentity: Identity,
+        method: String,
+        payload: String,
+        responseTimeoutMs: Int = 10000,
```

Comment:
The duration parameters in the RPC feature should be more platform-native. I think in this case you should either use the kotlin native [Duration type](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.time/-duration/) type (introduced kotlin 1.6) or you should provide these as `Long` with the `Millis` suffix (e.g. `responseTimeoutMillis: Long = 10000`)